gen/enum.go

```diff
@@ -96,23 +96,32 @@ func enum(g Generator, spec *compile.EnumSpec) error {
 		<end>
 
 		<$v := newVar "v">
+		<$value := newVar "value">
 		// UnmarshalText tries to decode <$enumName> from a byte slice
 		// containing its name.
 		<- if .Spec.Items>
 		//
 		//   var <$v> <$enumName>
 		//   err := <$v>.UnmarshalText([]byte("<(index .Spec.Items 0).Name>"))
 		<- end>
-		func (<$v> *<$enumName>) UnmarshalText(value []byte) error {
-			switch string(value) {
+		func (<$v> *<$enumName>) UnmarshalText(<$value> []byte) error {
+			<- $s := newVar "s" ->
+			<- $val := newVar "val" ->
+			<- $err := newVar "err">
```

Comment:
nit: don't need `newVar` for err since the convention is to always shadow `err`. The newVar business tries to ensure that we don't have two templated variables named "val", "s", or something conflicting with an import in this scope.